File: datestructure/java/Interview-1-ArraysAndString/IsUnique.java
/**
 * Is Unique: Implement an algorithm to determine if a string has all unique charaters,
 * What if you cannot use additional data structure.
 **/

 public class IsUnique {

    public static void main(String[] args) {

        String str = "abcdefghijklmn,.";

        if (isUniqueChars(str) == true) {
            System.out.println("This string is unique.");
        } else {
            System.out.println("This string is not unique.");
        }
    }

    static boolean isUniqueChars(String str) {
        if( str.length() > 128) {  
            return false;
        }

        boolean[] mark = new boolean[128];

        for (int i = 0; i < str.length(); i++) {
            int strIth = (int) str.charAt(i);
            if (mark[strIth]) {
                return false;
            }
            mark[strIth] = true;
        }
        return true;
    }

    
 }

File: datestructure/javascript/interview-1-ArrayAndString/isUnique.js
str = 'abcdefghijk,.';

var isUnique = function (str) {
    if(str.length > 128) {
        return false;
    }

    for (i = 0; i < str.length; i++) {
        for (j = 0; j <  str.length; j++) {
            if (str[i] == str[j]) {
                return false
            }
        }
    }
    return true

}
console.log(str.length)
console.log(isUnique(str))